src/benchmarklib/benchmark_runner.cpp

```diff
@@ -462,7 +462,8 @@ void BenchmarkRunner::_create_report(std::ostream& stream) const {
     report["segments"] = _sql_to_json("SELECT * FROM benchmark_segments_log");
   }
 
-  stream << std::setw(2) << report << std::endl;
+  // Write the output file
+  std::ofstream{_config.output_file_path.value()} << std::setw(2) << report << std::endl;
```

Comment:
You're writing the whole JSON report every time, right? Just to be sure I get this right.